common/jenkins-slaves/nodejs12/ocp-config/Tailorfile

```diff
@@ -2,4 +2,4 @@ namespace cd
 selector app=jenkins-slave-nodejs12
 param-file ../../../../../ods-configuration/ods-core.env
 ignore-unknown-parameters true
-bc,is
+template
```

Comment:
Please correct me if I am wrong, but you also use for `bc` and `is` a `template`
https://github.com/opendevstack/ods-quickstarters/blob/3a2c5f78be1e9b4d7dcb7ee27643f661df97e9de/common/jenkins-slaves/nodejs10-angular/ocp-config/bc.yml#L1-L3
https://github.com/opendevstack/ods-quickstarters/blob/3a2c5f78be1e9b4d7dcb7ee27643f661df97e9de/common/jenkins-slaves/nodejs10-angular/ocp-config/is.yml#L1-L3
shouldn't it be instead `kind: BuildConfig` and `kind: ImageStream`?

However, I don't know whats wrong to combine the two files into one to avoid inconsistency and double work e.g. regarding labeling and to take away the may unknown dependency for beginners when you first run the `buildConfig` before `imageStream` is available?

Reply:
It's fine to combine into one template, but you would still target the resources inside the template, not the template resource itself.

Please read https://github.com/opendevstack/tailor/blob/master/README.md to understand what Tailor does.

Reply:
Hi @michaelsauter, I am sorry but after read the README from tailor I still don't know whats the problem... So what's the story behind `bc,is` vs. `template` - I assumed that this would be referenced to the file name?

Reply:
From the readme:

> Any drift between your desired state (the YAML templates) and the current state (resources in the cluster) can be detected, reviewed and reconciled using a CLI interface. 

Tailor uses OpenShift Templates as input. A template can be processed, which returns a list of resources. A resource is e.g. a BuildConfig (bc) or an ImageStream(is).

Again from the readme:

> By default, all resource types are compared, but you can limit to specific ones, e.g. `diff pvc,dc`.

The argument to the `diff` command that you see here (`pvc,dc`) is, in your example, defined in the `Tailorfile` (see https://github.com/opendevstack/tailor#tailorfile).

Which means: `bc,is` refers to resource types, not file names. To complicate things a little, a `template` can also be a valid OpenShift resource itself. But you would need to declare a template which contains a template to make use of that ....

You only want to define a BuildConfig and an ImageStream in a template, and target the BuildConfig and ImageStream from the resulting list - because you want to compare the `BuildConfig` expressed in your template with the `BuildConfig` that exists in OpenShift.

Reply:
Gotcha! I changed it back 👍